paasta_tools/cli/cmds/spark_run.py

```diff
@@ -154,47 +160,76 @@ def get_docker_run_cmd(
+
+    if cmd == 'jupyter':
+        spark_env['SPARK_OPTS'] = spark_conf
+        spark_env['JUPYTER_RUNTIME_DIR'] = DEFAULT_SPARK_WORK_DIR + '/.jupyter'
+        spark_env['JUPYTER_DATA_DIR'] = DEFAULT_SPARK_WORK_DIR + '/.jupyter'
+
+    return spark_env
+
+
+def get_spark_conf_str(
     args,
     container_name,
     spark_ui_port,
     docker_img,
     system_paasta_config,
+    volumes,
 ):
-    spark_conf = {}
-    spark_conf['APP_NAME'] = container_name
-    spark_conf['SPARK_UI_PORT'] = spark_ui_port
-
-    creds = Session().get_credentials()
-    spark_conf['AWS_ACCESS_KEY_ID'] = creds.access_key
-    spark_conf['AWS_SECRET_ACCESS_KEY'] = creds.secret_key
+    spark_conf = list()
+    spark_conf.append('--conf spark.app.name=%s' % container_name)
+    spark_conf.append('--conf spark.ui.port=%d' % spark_ui_port)
 
     cluster_fqdn = system_paasta_config.get_cluster_fqdn_format().format(cluster=args.cluster)
     mesos_address = '{}:{}'.format(
         find_mesos_leader(cluster_fqdn),
         MESOS_MASTER_PORT,
     )
-    spark_conf['SPARK_MASTER'] = 'mesos://%s' % mesos_address
-    spark_conf['SPARK_CORES_MAX'] = args.max_cores
-    spark_conf['SPARK_EXECUTOR_CORES'] = args.executor_cores
-    spark_conf['SPARK_EXECUTOR_MEMORY'] = '%dg' % args.executor_memory
+    spark_conf.append('--conf spark.master=mesos://%s' % mesos_address)
+
+    spark_conf.append('--conf spark.cores.max=%d' % args.max_cores)
+    spark_conf.append('--conf spark.executor.memory=%dg' % args.executor_memory)
+    spark_conf.append('--conf spark.executor.cores=%d' % args.executor_cores)
 
     if args.driver_max_result_size:
-        spark_conf['SPARK_DRIVER_MAX_RESULT_SIZE'] = '%dg' % args.driver_max_result_size
+        spark_conf.append('--conf spark.driver.maxResultSize=%dg' % args.driver_max_result_size)
     if args.driver_memory:
-        spark_conf['SPARK_DRIVER_MEMORY'] = '%dg' % args.driver_memory
+        spark_conf.append('--conf spark.driver.memory=%dg' % args.driver_memory)
     if args.driver_cores:
-        spark_conf['SPARK_DRIVER_CORES'] = args.driver_cores
+        spark_conf.append('--conf spark.driver.cores=%d' % args.driver_cores)
 
-    if args.build:
-        spark_conf['SPARK_EXECUTOR_IMAGE'] = docker_img
+    spark_conf.append('--conf spark.mesos.executor.docker.image=%s' % docker_img)
+    if not args.build:
+        spark_conf.append('--conf spark.mesos.uris=file:///root/.dockercfg')
+
+    # TODO PAASTA-13815
+    # spark_conf.append('--conf spark.mesos.secret=')
+    # spark_conf.append('--conf spark.mesos.principal=')
+
+    # derby.system.home property defaulting to '.',
+    # which requires directory permission changes.
+    spark_conf.append('--conf spark.driver.extraJavaOptions=-Dderby.system.home=/tmp/derby')
 
-    return spark_conf
+    spark_conf.append('--conf spark.mesos.constraints=pool:%s' % args.pool)
+
+    spark_conf.append('--conf spark.mesos.executor.docker.volumes=%s' % ','.join(volumes))
```

Comment:
Does this work? I would be surprised this works without complex quoting

Reply:
Yeah, spark.mesos.executor.docker.volumes work fine without complex quoting, https://pb.yelpcorp.com/311715. If I don't set spark.mesos.executor.docker.volumes, this example will fail with ServiceKafkaDiscovery exceptions since the executor looks for KafkaDiscovery files.